dpypelines/s3_tar_received.py

```diff
@@ -1,22 +1,103 @@
+        assert len(directories) == 1, (
+            "Aborting, input directory has more than one directory within it"
+        )
+        decompressed_directory = directories[0]
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to identify decompressed directory in 'inputs'. Found directories: {directories}", err))
+        raise err
+
+    # Create a local directory store using the decompressed files
+    try:
+        localStore = LocalDirectoryStore(f"inputs/{decompressed_directory}")
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error(f"Failed to create local directory store at inputs/{decompressed_directory}", err))
+        raise err
+
+    # Verify if a pipeline configuration file ('pipeline-config.json') exists in the local directory store
+    try:
+        # Check for the existence of a configuration file
+        if not localStore.has_lone_file_matching("pipeline-config.json"):
+            msg = message.expected_local_file_missing("Missing pipeline-config.json", "pipeline-config.json", "start")
+            notify.data_engineering(msg)
+            # Create a default configuration file if one does not exist
+            default_config = {
+                "pipeline": "default_pipeline",
+                "parameters": {}
+            }
+            with open("pipeline-config.json", "w") as f:
+                json.dump(default_config, f)
+        else:
+            # Load the configuration file
+            config_dict = localStore.get_lone_matching_json_as_dict("pipeline-config.json")     
+    except Exception as err:
+        notify.data_engineering(message.unexpected_error("Error while checking for pipeline-config.json", err))
+        raise err
 
-    # decompress tar file to workspace
+    # Load the configuration file and validate it against a schema
+    try:
+        config_dict = localStore.get_lone_matching_json_as_dict("pipeline-config.json")
+    except Exception as err:
+        notify.data_engineering(message.pipeline_input_exception({"pipeline-config.json": "expected"}, localStore, err))
+        raise err
+    
+    # Retrieve the path to the schema for the configuration
+    try:
+        path_to_schema = get_config_schema_path(config_dict)
+    except Exception as err:
+        notify.data_engineering(message.cant_find_schema(config_dict, err))
+        raise err
 
-    # confirm we have a config
+    # Validate the configuration against the retrieved schema
+    try:    
+        validation.validate_json_schema(schema_path=path_to_schema, data_dict=config_dict)
+    except Exception as err:
+        notify.data_engineering(message.invalid_config(config_dict, err))
+        raise err
 
-    # if we dont have a config - make one? (tbd)
+    # Loads the pipeline specified in the configuration file and executes it with the provided parameters
```

Comment:
clever but don't worry about this just yet please (there's tasks to come on this). For now just explicitly call: https://github.com/ONSdigital/dp-data-pipelines/blob/0682f4e8ca72fc5309bbd032c56a85acceeedb92/dpypelines/pipeline/dataset_ingress_v1.py#L9 with the path to the directory (there's a method on the store to get it)